daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,599 @@
+		if err := f.removeIncomingRule(rule); err != nil {
+			return fmt.Errorf("removing incoming rules: %w", err)
+		}
+	}
+
+	// remove allow fileshare rules
+	for _, rule := range f.fileshareRules {
+		if err := f.commandExecutor.ExecuteCommand("-D " + rule); err != nil {
+			return fmt.Errorf("removing fileshare allow rule: %w", err)
+		}
+	}
+
+	f.enabled = false
+
+	return nil
+}
+
+func (f *FirewallManager) Enable() error {
+	if f.enabled {
+		return fmt.Errorf("firewall is already enabled")
+	}
+
+	// add traffic block
+	if f.trafficBlocked {
+		if err := f.addBlockTrafficRules(); err != nil {
+			return fmt.Errorf("blocking traffic: %w", err)
+		}
+	}
+
+	// add api allowlist
+	if err := f.manageApiAllowlist(true); err != nil {
+		return fmt.Errorf("adding api allowlist %w", err)
+	}
+
+	// add allowlist
+	for _, rule := range f.allowlistRules {
+		if err := f.commandExecutor.ExecuteCommand("-I" + rule); err != nil {
+			return fmt.Errorf("adding allowlist rule: %w", err)
+		}
+	}
+
+	// add meshnet block rules
+	if f.meshnetDeviceAddress != "" {
+		if err := f.addMeshnetBlockRules(f.meshnetDeviceAddress); err != nil {
+			return fmt.Errorf("adding meshnet block rules: %w", err)
+		}
+	}
+
+	// add allow incoming rules
+	for _, rule := range f.allowIncomingRules {
+		if err := f.addIncomingRule(rule); err != nil {
+			return fmt.Errorf("adding incoming rules: %w", err)
+		}
+	}
+
+	// add allow fileshare rules
+	for _, rule := range f.fileshareRules {
+		if err := f.commandExecutor.ExecuteCommand("-I " + rule); err != nil {
+			return fmt.Errorf("adding fileshare allow rule: %w", err)
+		}
+	}
+
+	f.enabled = true
+
+	return nil
+}
+
+// FileshareAllow adds ACCEPT rule for all incoming connections to tcp port 49111 from the peer with given UniqueAddress.
+func (f *FirewallManager) FileshareAllow(peer meshnet.UniqueAddress) error {
+	if _, ok := f.fileshareRules[peer.UID]; ok {
+		return fmt.Errorf("fileshare allow rule for the given peer already exists")
+	}
+
+	rule := fmt.Sprintf("INPUT -s %s/32 -p tcp -m tcp --dport 49111 -m comment --comment nordvpn -j ACCEPT", peer.Address.String())
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-I " + rule); err != nil {
+			return fmt.Errorf("adding fileshare allow rule: %w", err)
+		}
+	}
+
+	f.fileshareRules[peer.UID] = rule
+	return nil
+}
+
+// FileshareDeny removes ACCEPT rule for all incoming connections to tcp port 49111 from the peer with given UniqueAddress.
+func (f *FirewallManager) FileshareDeny(peer meshnet.UniqueAddress) error {
+	rule, ok := f.fileshareRules[peer.UID]
+	if !ok {
+		return fmt.Errorf("fileshare allow rule for the given peer doesn't exist")
+	}
+
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-D " + rule); err != nil {
+			return fmt.Errorf("removing fileshare allow rule: %w", err)
+		}
+	}
+
+	delete(f.fileshareRules, peer.UID)
+	return nil
+}
+
+func (f *FirewallManager) addBlockTrafficRules() error {
+	interfaces, err := f.devices()
```

Comment:
if a new interface is connected, e.g. usb adapter, will this be updated?

Reply:
New devices will be returned, but the rules themselves won't be updated. I think we should have a task for that, since this issue also exists on main.